File: src/main/java/com/example/homework/news/NewsContract.java
package com.example.homework.news;

import com.example.homework.data.Translation;

public interface NewsContract {
    interface NewsUIView {
        void onError();

        void initItems(Translation translation);

        //        void refreshItems(List<Translation.StoriesBean> dataSet, List<Translation.TopStoriesBean> topDataSet, List<View> viewList);
        void loadUpItems(Translation translation);
    }

    interface NewsPresenter {
        void initData();

        void refreshData();

        void onLoadMoreData();
    }
}

File: main/java/com/example/homework/news/NewsContract.kt
package com.example.homework.news

import com.example.homework.data.Translation

interface NewsContract {
    interface NewsUIView {
        fun onError()

        fun initItems(translation: Translation)

        fun refreshItems(translation: Translation)

        fun loadUpItems(translation: Translation)
    }

    interface NewsPresenter {
        fun initData()

        fun refreshData()

        fun onLoadMoreData()
    }
}
